contracts/src/v0.8/functions/dev/1_0_0/FunctionsBilling.sol

```diff
@@ -289,8 +300,8 @@ abstract contract FunctionsBilling is Routable, IFunctionsBilling {
 
     // No lower bound on the requested gas limit. A user could request 0
     // and they would simply be billed for the gas and computation.
-    if (billing.callbackGasLimit > s_config.maxCallbackGasLimit) {
-      revert GasLimitTooBig(billing.callbackGasLimit, s_config.maxCallbackGasLimit);
+    if (billing.callbackGasLimit > _getMaxCallbackGasLimit(billing.flags)) {
+      revert GasLimitTooBig(billing.callbackGasLimit, _getMaxCallbackGasLimit(billing.flags));
```

Comment:
Just want to advocate again that I think all flags based logic should be on the Router.
I would move this check on 303 to `_sendRequest` and the config to be in FunctionsRouter.

Conceptually does the DON care about callbackGasLimits? It doesn't seem so to me.

Only tricky part will be fighting with contract size limits, but we _could_ likely figure that out.